packages/component-base/src/data-provider-controller/cache.js

```diff
@@ -136,6 +137,44 @@ export class Cache {
     return this.flatSize;
   }
 
+  /**
+   * The number of items.
+   *
+   * @return {number}
+   */
+  get size() {
+    return this.__size;
+  }
+
+  /**
+   * Sets the number of items.
+   *
+   * @param {number} size
+   */
+  set size(size) {
+    const oldSize = this.__size;
+    if (oldSize === size) {
+      return;
+    }
+
+    this.__size = size;
+
+    if (this.context.placeholder !== undefined) {
+      this.items.length = size || 0;
+      for (let i = 0; i < size || 0; i++) {
+        // eslint-disable-next-line logical-assignment-operators
+        this.items[i] = this.items[i] || this.context.placeholder;
```

Comment:
The `||=` operator is already used in `vaadin-grid-helpers.js` and `vaadin-grid-pro-edit-select-mixin.js` so it should be safe

Reply:
Thanks, good to know. Updated.